app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,17 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      params = { format: 'html', template_id: ReportTemplate.find_by(name: "Subscription - Entitlement Report").id , input_values: { TemplateInput.find_by_name("Days from Now").id.to_s  => { value: UserMailNotification.first.mail_query.to_s } } }
+
+      composer = ReportComposer.new(params)
+      job = composer.schedule_rendering
+
+      base_url = report_data_report_template_path(ReportTemplate.find_by(name: "Subscription - Entitlement Report").id)
```

Comment:
It's been a while but I think you can specify job_id as a hash parameter to `report_data_report_template_path` to avoid a bit of the string interpolation at the next line